File: budget_app/budget_app/Model/Wallet.swift
//
//  Wallet.swift
//  budget_app
//
//  Created by Alina Novikova on 24/02/2024.
//

import Foundation

struct Wallet: Identifiable, Codable, Hashable {
    var id: Int
    var userID: Int
    var name: String
    var currency: String
    var initialBalance: Int
    var balance: Int
    var iconCode: String
    var colorCode: String
    var createdAt: String
    
    static var placeholder: Wallet {
        return Wallet(id: -1, userID: -1, name: "Select wallet", currency: "USD", initialBalance: 0, balance:0, iconCode: "questionmark.circle", colorCode: "#DBF9F0", createdAt: "")
    }
}

struct WalletStatistics: Identifiable, Codable {
    var id: Int
    let totalIncome: Int
    let totalExpenses: Int
}

File: backend/models/wallet.go
package models

import "time"

type Wallet struct {
	ID             int64     `json:"id" db:"id"`
	UserID         int64     `json:"userID" db:"userid"`
	Name           string    `json:"name" db:"name"`
	Currency       string    `json:"currency" db:"currency"`
	InitialBalance int       `json:"initialBalance" db:"initialbalance"`
	Balance        int       `json:"balance" db:"balance"`
	Icon           string    `json:"iconCode" db:"icon"`
	Color          string    `json:"colorCode" db:"color"`
	CreatedAt      time.Time `json:"createdAt" db:"createdat"`
}
